pkg/nvmeclient/nvme_client.go

```diff
@@ -495,6 +497,24 @@ func ConnectAll(discoveryRequest *hostapi.DiscoverRequest, maxIOQueues int) ([]*
 	return ctrls, nil
 }
 
+func connectNVMEDevicesWithRetry(request *ConnectRequest) (*CtrlIdentifier, error) {
+	var err error
+	var ctrlID *CtrlIdentifier
+
+	logrus.Debug("try to reconnect nvme-devices")
+
+	err = retry.Do(func() error {
```

Comment:
do you really need to do this? (as you use the same err inside - it is already set)

Reply:
AFAICT this is not the same error. I return the error from internal function and catch it in `connectNVMEDevicesWithRetry`

Reply:
actually I get a different result please see:
https://go.dev/play/p/R6HwBVTEFUV

Reply:
It's not really different, it (the err output from the retry itself) just logs all the errors you got - do you want to log all of them, or just the last one?